sdk/src/main/scala/com/horizen/account/api/rpc/service/EthService.scala

```diff
@@ -1,112 +1,178 @@
+      //case "safe" => //-39001: Unkown block
+      case "latest" => blockId = history.getBlockIdByHeight(nodeView.history.getCurrentHeight)
+      case "pending" => blockId = history.getBlockIdByHeight(nodeView.history.getCurrentHeight)
+      case _ => blockId = history.getBlockIdByHeight(Integer.parseInt(Numeric.cleanHexPrefix(tag), 16))
+    }
+    if (blockId.isEmpty) return null
+    blockId.get()
+  }
+
+  private def getBalance(address: String, tag: Quantity) : String = {
+    val stateDbRoot = getStateDbRootFromBlockById(getBlockIdByTag(tag.getValue))
+    if (stateDbRoot == null) return "null"
+    val balance = stateDbRoot.getBalance(Numeric.hexStringToByteArray(address))
+    Numeric.toHexStringWithPrefix(balance)
+  }
+
   @RpcMethod("eth_getTransactionCount") def getTransactionCount(address: String, tag: Quantity): Quantity = {
-    val nonce = nodeView.state.getNonce(Numeric.hexStringToByteArray(address))
+    val stateDbRoot = getStateDbRootFromBlockById(getBlockIdByTag(tag.getValue))
+    if (stateDbRoot == null) return new Quantity("null")
+    val nonce = stateDbRoot.getNonce(Numeric.hexStringToByteArray(address))
     new Quantity(Numeric.toHexStringWithPrefix(nonce))
   }
 
-  @RpcMethod("net_version") def version: String = Numeric.toHexStringWithPrefix(BigInteger.TEN) // TODO: sidechainSettings.genesisData.mcNetwork or just a number????
+  @RpcMethod("net_version") def version: String = String.valueOf(networkParams.chainId)
 
-  @RpcMethod("eth_estimateGas") def estimateGas(transaction: util.LinkedHashMap[String, String]/*, tag: Quantity*/) = { // TODO: We need an estimateGas function to execute EVM and get the gasUsed
-    if(transaction.containsKey("data") && transaction.get("data") != null && transaction.get("data").length > 1)
+  @RpcMethod("eth_estimateGas") def estimateGas(transaction: util.LinkedHashMap[String, String] /*, tag: Quantity*/) = { // TODO: We need an estimateGas function to execute EVM and get the gasUsed
+    if (transaction.containsKey("data") && transaction.get("data") != null && transaction.get("data").length > 1)
       new Quantity(Numeric.toHexStringWithPrefix(BigInteger.valueOf(3000000))) // particular smart contract creation
     else
       new Quantity(Numeric.toHexStringWithPrefix(BigInteger.valueOf(21000))) // EOA to EOA
   }
 
   @RpcMethod("eth_gasPrice") def gasPrice = { // TODO: Get the real gasPrice later
-    new Quantity("0x20")
+    new Quantity("0x1")
   }
 
-  @RpcMethod("eth_getTransactionByHash") def getTransactionByHash(transactionHash: Quantity): ResponseObject = {
-    val tx = getTransaction(transactionHash)
-    if (tx.isEmpty) return null
-    val ethTx = getEthereumTransaction(tx)
-    new ResponseObject(getTxObject(transactionHash, ethTx, tx))
+  @RpcMethod("eth_getTransactionByHash") def getTransactionByHash(transactionHash: String): EthereumTransactionView = {
+    val tx = nodeView.history.searchTransactionInsideBlockchain(Numeric.cleanHexPrefix(transactionHash))
```

Comment:
this method is too costly.
Instead we should:
1) for given tx hash get the receipt from the state
2) extract from receipt block id and tx index
3) get block by id from history and extract tx for given index.